mesa/space.py

```diff
@@ -1033,11 +1034,19 @@ def place_agent(self, agent: Agent, node_id: int) -> None:
         self.G.nodes[node_id]["agent"].append(agent)
         agent.pos = node_id
 
-    def get_neighbors(self, node_id: int, include_center: bool = False) -> list[int]:
-        """Get all adjacent nodes"""
-        neighbors = list(self.G.neighbors(node_id))
-        if include_center:
-            neighbors.append(node_id)
+    def get_neighbors(
+        self, node_id: int, include_center: bool = False, radius: int = 1
+    ) -> list[int]:
+        """Get all adjacent nodes within a certain radius"""
+        if radius == 1:
+            neighbors = list(self.G.adj[node_id])
+            if include_center:
+                neighbors.append(node_id)
+        else:
+            neighbors = nx.single_source_shortest_path_length(self.G, node_id, radius)
+            if not include_center:
+                del neighbors[node_id]
+            neighbors = list(neighbors)
```

Comment:
`neighbors` was a dict, then you only keep the keys here? The Mypy annotation is going to be messy. Should be call the `dict` `neighbors` a different name?

Reply:
changed the name in `neighbors_with_distance` for the dict